src/components/Articles.jsx

```diff
@@ -1,15 +1,51 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { getTopics } from "../utils/api";
 import { Articlelist } from "./Articlelist";
 import "./Articles.css"
 
 export const Articles = () => {
+
+    const [topics, setTopics] = useState([])
+    const [topic, setTopic] = useState('')
+    const [loading, Isloading] = useState(true)
+
+
+    useEffect(() => {
+        getTopics()
+            .then((topics) => {
+                setTopics(topics)
+                Isloading(false)
+            })
+    }, [])
+
+
+    if (loading) {
+        return <p>Loading...</p>
+    }
+
     return (
-        <div className="List">
-            <div className="col"></div>
-            <div className="Content">
-                <Articlelist />
+        <div>
+            <div className="List">
+                <div className="col"></div>
+                <div className="Content">
+                    <div className="topics">
+                        {topics.map((topic) => {
+                            return (
+
+                                <button key={topic.slug} className="button" onClick={() => {
+                                    setTopic(topic.slug)
+                                }}>{topic.slug}</button>
+
+                            )
+                        })}
+                        <button className="button" onClick={() => {
+                            setTopic('')
```

Comment:
It would be nice for the user to be able to click the 'Articles' link in the nav bar and see 'all articles' again (reset the filtering) this might come into action when you start making the URL reflect the topic you're filtering by.